micrometer-tracing-bridges/micrometer-tracing-bridge-brave/src/main/java/io/micrometer/tracing/brave/bridge/W3CPropagation.java

```diff
@@ -16,7 +16,6 @@
 package io.micrometer.tracing.brave.bridge;
 
 import brave.internal.baggage.BaggageFields;
```

Comment:
this is also an internal type, but not being removed so far.. maybe by accident ;)

Reply:
p.s I'm not sure if this was copied because brave's wasn't published to maven central. It is now, so maybe can reduce maintenance here https://central.sonatype.com/artifact/io.zipkin.contrib.brave-propagation-w3c/brave-propagation-tracecontext/versions from https://github.com/openzipkin-contrib/brave-propagation-w3c

Reply:
Actually I think I copied this code from OTel and adopted it to Brave. It was a long time ago so I don't really remember :shrug:

Reply:
could have been. in any case this variant was done a while back benchmarked etc to be same perf as the built-in b3 stuff. hopefully it helps

Reply:
yeah definitely was otel, not just your comments say that, but the impl is wrong!

Reply:
here's a PR to re-use the code I wrote a few years ago, highlighting the uberbug in the current impl https://github.com/micrometer-metrics/tracing/pull/532